### DISCUSSION TITLE
Listings with unrendered documents

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#7188] [Q&A] [answered]
### Description

I have a website that features a [listing of 11 documents](https://www.stat20.org/assignments#problem-sets). They appear using a [custom ejs](https://github.com/stat20/stat20/blob/main/assets/listings/assignments-ps.ejs) template that pulls from the title of each document to make the text for a link. Here's the template:

```html
<ul>
<% for (let i = 0; i < items.length; i++) { %>
  <li><a href="<%- items[i].path %>">PS <%= i+1 %>: <%= items[i].title %></a></li>
<% } %>
</ul>
```
The listing renders like this:

<img width="317" alt="Screen Shot 2023-10-10 at 9 28 07 AM" src="https://github.com/quarto-dev/quarto-cli/assets/1627362/93cb3881-85f5-4a99-b6dc-886c41ac85cb">

The difference between the PS that appear correctly in the listing and those that do not is that the former render to pdf and the latter are [ignored during render](https://github.com/stat20/stat20/blob/62bc790fc63c1a5a024d3284e67038e71ec4b2fd/_quarto.yml#L5-L10) in `_quarto.yml`. This was the only way that I was able to a) provide the document to students as a raw qmd that they could upload or copy/paste into RStudio while b) getting it to show up in in the listing.

Are there any suggestions for how to solve this in a way that will not partially break the listing? This is a version of a general issue that I've come up with in a few contexts: how to make a qmd file accessible from a website? I can hardcode links to a github raw file, but I'd rather not.

Thanks for the help!

> Quarto version 1.3.433
> MacOS 12.6.8

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by andrewpbray** [reply]
Sure thing. Please see attached.

[listing-with-unrendered.zip](https://github.com/quarto-dev/quarto-cli/files/12859923/listing-with-unrendered.zip)

**Reply by andrewpbray** [reply]
@mcanouil Please let me know if his reprex didn't work for you. Happy to remake it.

**Reply by mcanouil** [reply]
Sorry, did not had time to look into it at this time.
(also why providing those reprex from the start helps)

**Reply by mcanouil** [reply, marked as answer]
If you want to keep the Quarto documents for your students, use code tools to make it part of the documents.
<https://quarto.org/docs/output-formats/html-code.html#code-tools>

Also you can linked formats, see <https://quarto.org/docs/output-formats/html-multi-format.html#specifying-formats-to-link>.

- `_quarto.yml`

```yaml
project:
  type: website

website:
  title: "listing-with-unrendered"
  navbar:
    left:
      - href: index.qmd
        text: Home
      - problem-sets.qmd

format:
  html:
    theme: cosmo
    css: styles.css
    toc: true
```

- `problem-sets.qmd`

````qmd
---
title: "Problem Sets"
listing:
- id: ps-listing
  template: assignments-ps.ejs
  contents: posts
  sort: "date desc"
  type: table
  fields:
    - title
  categories: false
  sort-ui: false
  filter-ui: false
---

Here is the listing:

::: {#ps-listing}
:::
````

- `posts/ps-1.qmd`

````qmd
---
title: "Intro Probability"
format:
  html: default
  pdf: default
format-links: [pdf]
---

About this site

```{r}
1 + 1
```
````


https://github.com/quarto-dev/quarto-cli/assets/8896044/befed282-584a-4b5b-beed-241c62063b33

**Reply by andrewpbray** [reply]
I hadn't thought of code tools! That works just swell, especially when combined with upstream metadata so that the code tool yaml doesn't get copied into the source for the new doc.
 
 Thanks!
